workspace-manager-clienttests/src/main/java/scripts/utils/ResourceMaker.java

```diff
@@ -0,0 +1,172 @@
+            .metadata(
+                new ReferenceResourceCommonFields()
+                    .cloningInstructions(CloningInstructionsEnum.NOTHING)
+                    .description("Descriptiono of " + name)
+                    .name(name))
+            .bucket(new GcpGcsBucketAttributes().bucketName(TEST_BUCKET_NAME));
+
+    return resourceApi.createBucketReference(body, workspaceId);
+  }
+
+  public static GcpGcsBucketResource makeControlledGcsBucketUserShared(
+      ControlledGcpResourceApi resourceApi, UUID workspaceId, String name) throws Exception {
+
+    // TODO: we should tolerate getting no default storage and no lifecycle rule. That is captured
+    //  in PF-635. For now, we build those things so the bucket gets created.
+    GcpGcsBucketLifecycleRule lifecycleRule =
+        new GcpGcsBucketLifecycleRule()
+            .action(
+                new GcpGcsBucketLifecycleRuleAction()
+                    .type(
+                        GcpGcsBucketLifecycleRuleActionType
+                            .DELETE)) // no storage class required for delete actions
+            .condition(
+                new GcpGcsBucketLifecycleRuleCondition()
+                    .age(64)
+                    .live(true)
+                    .addMatchesStorageClassItem(GcpGcsBucketDefaultStorageClass.ARCHIVE)
+                    .numNewerVersions(2));
+
+    List<GcpGcsBucketLifecycleRule> lifecycleRules = new ArrayList<>(List.of(lifecycleRule));
+
+    String bucketName = ClientTestUtils.getTestBucketName();
+    String jobId = UUID.randomUUID().toString();
+    var body =
+        new CreateControlledGcpGcsBucketRequestBody()
+            .common(
+                new ControlledResourceCommonFields()
+                    .accessScope(AccessScope.SHARED_ACCESS)
+                    .managedBy(ManagedBy.USER)
+                    .cloningInstructions(CloningInstructionsEnum.NOTHING)
+                    .description("Description of " + name)
+                    .name(name)
+                    .jobControl(new JobControl().id(jobId)))
+            .gcsBucket(
+                new GcpGcsBucketCreationParameters()
+                    .name(bucketName)
+                    .defaultStorageClass(GcpGcsBucketDefaultStorageClass.STANDARD)
+                    .lifecycle(new GcpGcsBucketLifecycle().rules(lifecycleRules))
+                    .location("US-CENTRAL1"));
+
+    logger.info("Creating bucket {} jobId {} workspace {}", bucketName, jobId, workspaceId);
+    CreatedControlledGcpGcsBucket createdBucket = resourceApi.createBucket(body, workspaceId);
+    while (ClientTestUtils.jobIsRunning(createdBucket.getJobReport())) {
+      TimeUnit.SECONDS.sleep(CREATE_BUCKET_POLL_SECONDS);
+      createdBucket = resourceApi.getCreateBucketResult(workspaceId, jobId);
+    }
+    logger.info("Create bucket status is {}", createdBucket.getJobReport().getStatus().toString());
+    if (createdBucket.getJobReport().getStatus() != JobReport.StatusEnum.SUCCEEDED) {
```

Comment:
can we use `assert()` in this kind of test?

Reply:
Some lightweight assertion library that would throw an assertion failure in the test seems helpful. I suspect others will have opinions here.